OpenKeychain/src/main/java/org/sufficientlysecure/keychain/keyimport/HkpKeyserverClient.java

```diff
@@ -102,19 +99,33 @@
      * seconds since 1/1/1970 UTC time)</li>
      * <li>%<b>flags</b>% = letter codes to indicate details of the key, if any. Flags may be in any
      * order. The meaning of "disabled" is implementation-specific. Note that individual flags may
-     * be unimplemented, so the absence of a given flag does not necessarily mean the absence of
-     * the detail.
+     * be unimplemented, so the absence of a given flag does not necessarily mean the absence of the
+     * detail.
      * <ul>
      * <li>r == revoked</li>
      * <li>d == disabled</li>
      * <li>e == expired</li>
      * </ul>
      * </li>
      * </ul>
+     *
+     * @see <a href="http://tools.ietf.org/html/draft-shaw-openpgp-hkp-00#section-5.2">
+     * 5.2. Machine Readable Indexes</a>
+     * in Internet-Draft OpenPGP HTTP Keyserver Protocol Document
      */
-    private static final Pattern UID_LINE = Pattern
-            .compile("uid:([^:]*):([0-9]+):([0-9]*):([rde]*)",
-                    Pattern.CASE_INSENSITIVE);
+    private static final Pattern PUB_KEY_LINE = Pattern
+            .compile( "(?<pub>pub:" +                 // group 1
+                            "(?<pubKeyID>[0-9a-fA-F]+)" +   // group 2
```

Comment:
you can use non-capturing groups here if they're not needed. also, why use named groups without referencing them later? `(?:XXX)`

Reply:
ah, I see. alright then :)

with non-capturing groups you can avoid capturing all parenthesis, so you don't need to skip numbers (2, 3, 4 instead of 2, 4, 6)